packages/core/src/agent/Dispatcher.ts

```diff
@@ -45,15 +48,32 @@ class Dispatcher {
     try {
       outboundMessage = await handler.handle(messageContext)
     } catch (error) {
-      this.logger.error(`Error handling message with type ${message.type}`, {
-        message: message.toJSON(),
-        error,
-        senderVerkey: messageContext.senderVerkey,
-        recipientVerkey: messageContext.recipientVerkey,
-        connectionId: messageContext.connection?.id,
-      })
-
-      throw error
+      const problemReportMessage = error.problemReport
+
+      if (
+        (problemReportMessage instanceof ConnectionProblemReportMessage ||
+          problemReportMessage instanceof CredentialProblemReportMessage ||
+          problemReportMessage instanceof PresentationProblemReportMessage) &&
+        messageContext.connection
+      ) {
```

Comment:
If we check if the error is an instance of `ProblemReportError` we can generically handle it for all errors that extend from the `ProblemReportError`. this way we don't have to check for specific message type in the dispatcher

```suggestion
      if (problemReportMessage instanceof ProblemReportError && messageContext.connection) {
          const problemReportMessage = error.problemReport
```